app/Helpers/CampForum.php

```diff
@@ -336,4 +336,51 @@ public static function sendEmailToSupportersForumPost($topicid, $campnum, $link,
             }
         }
     }
+
+    public static function getThreadLogUsers($topicid, $campnum)
+    {
+        $subscriptionUser = [];
+        $userExist = [];
+        $userIds= [];
+        $subUser=[];
+        $filter['topicNum'] = $topicid;
+        $filter['asOf'] = '';
+        $filter['campNum'] = $campnum;
+        $camp = CampForum::getForumLiveCamp($filter);
+        $subCampIds = CampForum::getForumAllChildCamps($camp);
+        foreach ($subCampIds as $camp_id) {
+            $directSupporter = CampForum::getDirectCampSupporter($topicid, $camp_id);
```

Comment:
Possibly we can combine the $directSupporters and $subscribers query

Reply:
Can't we use `join` ? Just checking

Reply:
Actually within both of these two separate functions there are multiple queries running with different conditions based on different data. data is also processed inside these functions before and after query is executed. their is no such similarity between subscription and direct support based on which we can satisfy our need.